geopandas/sindex.py

```diff
@@ -754,23 +765,36 @@ def query_bulk(self, geometry, predicate=None, sort=False):
         def nearest(
             self, geometry, return_all=True, max_distance=None, return_distance=False
         ):
-            if not compat.PYGEOS_GE_010:
-                raise NotImplementedError("sindex.nearest requires pygeos >= 0.10")
-
-            geometry = self._as_geometry_array(geometry)
+            if not (compat.USE_SHAPELY_20 or compat.PYGEOS_GE_010):
+                raise NotImplementedError(
+                    "sindex.nearest requires shapely >= 2.0 or pygeos >= 0.10"
+                )
 
-            if not return_all and max_distance is None and not return_distance:
-                return self._tree.nearest(geometry)
+            # breakpoint()
 
-            result = self._tree.nearest_all(
-                geometry, max_distance=max_distance, return_distance=return_distance
-            )
+            geometry = self._as_geometry_array(geometry)
+            if isinstance(geometry, BaseGeometry) or geometry is None:
+                geometry = [geometry]
+
+            if compat.USE_SHAPELY_20:
+                result = self._tree.query_nearest(
+                    geometry,
+                    max_distance=max_distance,
+                    return_distance=return_distance,
+                    all_matches=return_all,
+                )
+            else:
+                if not return_all and max_distance is None and not return_distance:
+                    return self._tree.nearest(geometry)
+                result = self._tree.nearest_all(
+                    geometry, max_distance=max_distance, return_distance=return_distance
+                )
             if return_distance:
                 indices, distances = result
             else:
                 indices = result
 
-            if not return_all:
+            if not return_all and not compat.USE_SHAPELY_20:
```

Comment:
```suggestion
            if not (return_all and compat.USE_SHAPELY_20):
```

Reply:
I am getting a headache of reasoning about all this version options and and/or/not combinations .. This change turned out to be wrong

(`not (False and True)` is not the same as `(not False and not True)`)